types/mydecimal.go

```diff
@@ -1227,10 +1228,13 @@ func (d *MyDecimal) WriteBin(precision, frac int, buf []byte) ([]byte, error) {
 		}
 	}
 
-	if fracSize < fracSizeFrom {
+	if fracSize < fracSizeFrom ||
+		(fracSize == fracSizeFrom && (trailingDigits <= trailingDigitsFrom || wordsFrac <= wordsFracFrom)) {
```

Comment:
Could you add some tests when `trailingDigits > trailingDigitsFrom` or  `wordsFrac > wordsFracFrom`?